### PR TITLE
Welcome Activity(Sliding page)

### PR DESCRIPTION
**Pull request by surajprakashpatel**
Welcome Activity

### CODE REVIEW

File: app/src/main/java/com/zariya/zariya/auth/presentation/fragment/WelcomeFragment.kt [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        initialization()
+    }
+
+    private fun initialization() {
```
Comment (VivekSharma811) [inline app/src/main/java/com/zariya/zariya/auth/presentation/fragment/WelcomeFragment.kt:39]: Rename function name to more self explanatory one

File: app/src/main/java/com/zariya/zariya/auth/presentation/fragment/WelcomeFragment.kt [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            R.layout.welcome_slide3,
+            R.layout.welcome_slide4
+        )
+
+        // adding bottom dots
+        addBottomDots(0)
+
+        //setup viewpager
+        myViewPagerAdapter = MyViewPagerAdapter()
+        binding!!.viewPager.adapter = myViewPagerAdapter
```
Comment (VivekSharma811) [inline app/src/main/java/com/zariya/zariya/auth/presentation/fragment/WelcomeFragment.kt:53]: remove !! after binding, as binding can't be null

File: app/src/main/java/com/zariya/zariya/auth/presentation/fragment/WelcomeFragment.kt [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                binding!!.btnSkip.setTextColor(resources.getColor(R.color.dark_blue))
+                binding!!.btnNext.setTextColor(resources.getColor(R.color.white))
+            } else {
+                binding!!.btnNext.setTextColor(Color.WHITE)
+            }
+            // changing the next button text 'NEXT' / 'GOT IT'
+            if (position == layouts.size - 1) {
+                // last page. make button text to GOT IT
+                binding!!.btnNext.setTextColor(resources.getColor(R.color.white))
+                binding!!.btnNext.text = "Start"
```
Comment (VivekSharma811) [inline app/src/main/java/com/zariya/zariya/auth/presentation/fragment/WelcomeFragment.kt:132]: move Start to strings.xml

File: app/src/main/res/layout/fragment_welcome.xml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".auth.presentation.fragment.WelcomeFragment">
+
+    <androidx.viewpager.widget.ViewPager
+        android:id="@+id/view_pager"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent" />
+
+    <TextView
```
Comment (VivekSharma811) [inline app/src/main/res/layout/fragment_welcome.xml:14]: use MaterialTextView instead

File: app/src/main/res/layout/fragment_welcome.xml [hunk 5]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (VivekSharma811) [inline app/src/main/res/layout/fragment_welcome.xml:2]: Better if we use Constraint layout

File: app/src/main/res/layout/welcome_slide1.xml [hunk 6]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (VivekSharma811) [inline app/src/main/res/layout/welcome_slide1.xml:2]: Use Constraint layout instead of nested layouts

**Review by VivekSharma811 — DISMISSED** [review]

**Review by Mahima081 — APPROVED** [review]
Approved.

**Review by VivekSharma811 — APPROVED** [review]

**Review by Mahima081 — APPROVED** [review]
LGTM

**Pull request merged**